pydarn/plotting/fan.py

```diff
@@ -185,11 +183,28 @@ def plot_fan(cls, dmap_data: List[dict], ax=None,
         	date = scan_time
 
         # Plot FOV outline
-        if ranges is None:
-            ranges = [0, dmap_data[0]['nrang']]
+        if ranges == [] or ranges is None:
+            try:
+                # If not given, get ranges from data file
+                ranges = [0, dmap_data[0]['nrang']]
+            except KeyError:
```

Comment:
I guess this is the same as the comment on line 302, ideally it'll never happen, but just in case do we want to code to continue plotting with defaults or throw an error/warning?